services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Persistence.kt

```diff
@@ -134,8 +134,20 @@ class HankeEntity (
         var meluHaitta: Haitta13? = null
         var polyHaitta: Haitta13? = null
         var tarinaHaitta: Haitta13? = null
-}
 
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is HankeEntity) return false
+
+        if (hankeTunnus != other.hankeTunnus) return false
```

Comment:
Hmm.. must be careful here.  From e.g. database point of view, there can be (currently/soon) up to 3 different rows each with the same hankeTunnus. Also, there could be a row with no hankeTunnus (for autosave-purposes in future).
I do not know which all parties will be using these equals() and hashCode(), so I'd check also fields "id" and "saveType" to be on the safe side.